Adding bug fixes for Node, MCO, Insights Op, and Cloud Compute (#34194)
openshift/openshift-docs: release_notes/ocp-4-8-release-notes.adoc

* Previously, if you deleted a master machine using `oc delete machine` in an AWS cluster, the machine was not removed from the load balancers. As a result, the load balancer continued to serve requests to the removed master machine. With this fix, when you remove a master machine, the load balancer no longer serves requests to the machine. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1880757[*BZ#1880757*])

* Previously, when deleting an unreachable machine, the vSphere Virtual Machine Disk (VMDK) that is created for persistent volumes and attached to the node was incorrectly deleted. As a result, the data on the VMDK was unrecoverable. With this fix, the vSphere cloud provider checks for and detaches these disks from the node if the kubelet is not reachable. As a result, you can delete an unreachable machine without losing the VMDK. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1883993[*BZ#1883993*])

> * Previously, because a generated list of AWS instance types was out of date, some newer Amazon Web Services (AWS) instance types were not able to scale from zero when using the Cluster Autoscaler Operator and machine sets with zero replicas. The list of AWS instance types was updated to include newer instance types. With this fix, more instance types are available to the Cluster Autoscaler Operator for scaling from zero replicas. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1896321[*BZ#1896321*])

Reviewer: ```suggestion
* Previously, because a generated list of AWS instance types was out of date, some newer Amazon Web Services (AWS) instance types were not able to scale from zero when using the Cluster Autoscaler Operator and machine sets with zero replicas. The list of AWS instance types is now updated to include newer instance types. With this fix, more instance types are available to the Cluster Autoscaler Operator for scaling from zero replicas. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1896321[*BZ#1896321*])
```

*Insights Operator*

> * Previously, the Insights Operator did not collect Cluster Version Operator (CVO) pods or events in the `openshift-cluster-version` namespace. As a result, the Insights Operator did not display information about any problems that the CVO might experience. As a result, users could not get diagnostic information about the CVO. The Insights Operator was changed to gather the CVO pods and events from the `openshift-cluster-operator` namespace. As a result, issues with the the CVO are reorted by the Insights Operator. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942271[*BZ#1942271*])

Reviewer: ```suggestion
* Previously, the Insights Operator did not collect Cluster Version Operator (CVO) pods or events in the `openshift-cluster-version` namespace. As a result, the Insights Operator did not display information about any problems that the CVO might experience, and users could not get diagnostic information about the CVO. The Insights Operator is now updated to gather the CVO pods and events from the `openshift-cluster-operator` namespace so that issues with the the CVO are reported by the Insights Operator. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942271[*BZ#1942271*])
```
Just trying to avoid so many `As a result`s

*Machine Config Operator*

> * When upgrading from 4.6 to 4.7, the host name set by the *vsphere-hostname* service was applied only when a node is installed. If the host name was not statically set prior to upgrading, the host name could be lost. This update removes the condition which allowed the *vsphere-hostname* service to run only when a node is installed. As a result, vSphere host names are no longer lost when upgrading. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942207[*BZ#1942207*])

Reviewer: ```suggestion
* When upgrading from 4.6 to 4.7, the host name set by the *vsphere-hostname* service was applied only when a node was installed. If the host name was not statically set prior to upgrading, the host name could be lost. This update removes the condition which allowed the *vsphere-hostname* service to run only when a node is installed. As a result, vSphere host names are no longer lost when upgrading. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942207[*BZ#1942207*])
```
I thought a service name would be in backticks rather than bold, but I'm not sure.

> * Due to a bug in Keepalived 2.0.10, if a liveness probe kills a Keepalived container, any virtual IP addresses (VIPs) that were assigned to the system remain and are not cleaned up when Keepalived restarts. As a result, multiple nodes could hold the same VIP. Now, the VIPs are removed when Keepalived is started. As a result, VIPs are held by a single node. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1931505[*BZ#1931505*])

Reviewer: ```suggestion
* Due to a bug in Keepalived 2.0.10, if a liveness probe killed a Keepalived container, any virtual IP addresses (VIPs) that were assigned to the system remained and were not cleaned up when Keepalived restarted. As a result, multiple nodes could hold the same VIP. Now, the VIPs are removed when Keepalived is started. As a result, VIPs are held by a single node. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1931505[*BZ#1931505*])
```

> * Previously, empty static pod files were being written to the `/etc/kubernetes/manifests` directory. As a result, the kubelet log was reporting errors that could cause confusion with some users. Empty manifests are now moved to a different location when they are not needed. As a result, the errors are not appearing in the kubelet log. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1927042[*BZ#1927042*])

Reviewer: ```suggestion
* Previously, empty static pod files were being written to the `/etc/kubernetes/manifests` directory. As a result, the kubelet log was reporting errors that could cause confusion with some users. Empty manifests are now moved to a different location when they are not needed. As a result, the errors do not appear in the kubelet log. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1927042[*BZ#1927042*])
```

*Node*

* Previously, the CRI-O logs did not contain information about the source where images were pulled from. With this fix, the log pull source is added to the info level info-level CRI-O logs. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1881694[*BZ#1881694*])

> * Previously, when pods were created and deleted rapidly, a pod might not have enough time to complete the pod sandbox creation before the pod starts deletion. As a result, pod deletion could fail with a `ErrCreatePodSandbox`error. This error is now ignored if a pod is terminating. As a result, pod termination no longer fails if a pod could not complete the pod sandbox creation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1908378[*BZ#1908378*])

Reviewer: ```suggestion
* Previously, when pods were created and deleted rapidly, a pod might not have enough time to complete the pod sandbox creation before the pod started deletion. As a result, pod deletion could fail with a `ErrCreatePodSandbox` error. This error is now ignored if a pod is terminating. As a result, pod termination no longer fails if a pod could not complete the pod sandbox creation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1908378[*BZ#1908378*])
```

> * Previously, the Machine Config Operator (MCO) does not accept *trace* as a valid log level. As a result, the MCO cannot provide a method to enable trace-level logging, even though CRI-O supports it. The MCO was modified to support the *trace* log level. As a result, users can see a trace log level through the MCO configurations. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1930620[*BZ#1930620*])

Reviewer: ```suggestion
* Previously, the Machine Config Operator (MCO) did not accept *trace* as a valid log level. As a result, the MCO could not provide a method to enable trace-level logging, even though CRI-O supports it. The MCO is now updated to support the *trace* log level. As a result, users can see a trace log level through the MCO configurations. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1930620[*BZ#1930620*])
```

> * Previously, the kubelet tried to get the status of images that are not completely pulled. As a result, `crictl` reports a `error locating item named "manifest"` error for these images. CRI-O was changed to not list images that do not have a manifest. As a result, `crictl` no longer reports these errors. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942608[*BZ#1942608*])

Reviewer: ```suggestion
* Previously, the kubelet tried to get the status of images that are not completely pulled. As a result, `crictl` reports an `error locating item named "manifest"` error for these images. CRI-O is now updated to not list images that do not have a manifest. As a result, `crictl` no longer reports these errors. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1942608[*BZ#1942608*])
```

> * Previously, outdated status messages were not removed. Because of this, the Machine Config Operator might not be able to locate the proper machine config pool. A cleanup function was added to limit the number of statuses. As a result,  the MCO keeps at most 3 different kubeletConfig status. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1950133[*BZ#1950133*])

Reviewer: ```suggestion
* Previously, outdated status messages were not removed. Because of this, the Machine Config Operator (MCO) was sometimes unable to locate the proper machine config pool. With this release, a cleanup function is added to limit the number of statuses. As a result,  the MCO keeps at most 3 different kubeletConfig status. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1950133[*BZ#1950133*])
```

> * Previously, when upgrading from 4.6.25, in clusters with more than one `kubeletconfig` CR or `ContainerRuntimeConfig` CR, the Machine Config Operator (MCO) could generate duplicate machine configs for the same configuration. Consequently, the upgrade failed because the MCO would use the old controller version (IGNITIONVERSION 3.1.0). This update cleans up outdated duplicate machine configs and allows proper upgrading from 4.6.25. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1955517[*BZ#1955517*])

Reviewer: ```suggestion
* Previously, when upgrading from {product-title} version 4.6.25, in clusters with more than one `kubeletconfig` CR or `ContainerRuntimeConfig` CR, the Machine Config Operator (MCO) could generate duplicate machine configs for the same configuration. Consequently, the upgrade failed because the MCO would use the old controller version (IGNITIONVERSION 3.1.0). This update cleans up outdated duplicate machine configs and allows proper upgrading from version 4.6.25. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1955517[*BZ#1955517*])
```
